compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala

```diff
@@ -96,7 +96,8 @@ object QuoteDriver {
       case cl: URLClassLoader =>
         // Loads the classes loaded by this class loader
         // When executing `run` or `test` in sbt the classpath is not in the property java.class.path
-        val newClasspath = cl.getURLs.map(_.getFile())
+        import java.nio.file.Paths
+        val newClasspath = cl.getURLs.map(url => Paths.get(url.toURI).toFile)
```

Comment:
Why do we need a conversion to File here? It seems we are only interested in the String representation of these paths. I would make the conversion to String explicit. E.g.
```scala
val newClasspath = cl.getURLs.map(url => Paths.get(url.toURI).toString)
```

Reply:
Indeed, `toString` is better and I just tested it works on windows.

Reply:
I like Allan's suggestion and I will test it ASAP:
```
val newClasspath = cl.getURLs.map(url => Paths.get(url.toURI).toString)
```
For comparison here is my local change (*pending PR for several weeks now*):
```
val newClasspath = cl.getURLs.map(url => new JFile(url.getFile).getAbsolutePath)
```

Reply:
Thanks @michelou . It seems there are a lot of changes needed for the tests. I propose to restrict this PR for the compiler fixes, and make a separate PR for the test fixes.